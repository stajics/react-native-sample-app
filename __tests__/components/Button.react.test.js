import 'react-native';
import React from 'react';
import Button from '../../app/components/Button.react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'; // eslint-disable-line

it('renders not loading button', () => {
  const button = renderer.create(
    <Button
      text={'LOGIN'}
      onPress={jest.fn}
    />
  ).toJSON();
  expect(button).toMatchSnapshot();
});

it('renders isLoadng button', () => {
  const button = renderer.create(
    <Button
      text={'LOGIN'}
      onPress={jest.fn}
      isLoading
    />
  ).toJSON();
  expect(button).toMatchSnapshot();
});
