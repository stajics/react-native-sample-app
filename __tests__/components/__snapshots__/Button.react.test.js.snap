exports[`test renders isLoadng button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "black",
          "borderWidth": 1,
          "height": 20,
          "justifyContent": "center",
          "marginTop": 50,
          "width": 100,
        },
        undefined,
      ]
    }>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      } />
  </View>
</View>
`;

exports[`test renders not loading button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "black",
          "borderWidth": 1,
          "height": 20,
          "justifyContent": "center",
          "marginTop": 50,
          "width": 100,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 15,
          },
          undefined,
        ]
      }>
      LOGIN
    </Text>
  </View>
</View>
`;
